visidata/column.py

```diff
@@ -200,18 +207,13 @@ def fmtstr(self):
     def fmtstr(self, v):
         self._fmtstr = v
 
-    @property
-    def formatter(self):
-        return self._formatter.__name__[7:]  # remove leading 'format_'
-
-    @formatter.setter
-    def formatter(self, v: str):
-        self._formatMaker = getattr(col, 'format_'+(v or 'generic'))
-        self._formatter = col._formatMaker(self._format)
-
     def format_generic(self, fmtstr):
         return self.formatValue
 
+    def format(self, *args, **kwargs):
+        self._formatMaker = getattr(self, 'format_'+(self.formatter or 'generic'))
+        return self._formatMaker(self._formatdict)(*args, **kwargs)
```

Comment:
Now, `col.format` will build the `_formatMaker`, and `self.formatter` is a simple string.